Adding a phpunit test for contact query order by contact_sub_type:label

syxys1 opened the pull request:
Overview
----------------------------------------
_Adding a test to verify behavior of SK contact query when ordering by contact_sub_type:label.  See relevant conversation in 
Mattermost FormBuilder channel last september 25th. [https://chat.civicrm.org/civicrm/pl/gekpmfwj4b859mfkf6gjbrytie](url) ._
 
Before
----------------------------------------
_No test was present._

After
----------------------------------------
_Added test testGetWithOrderBy() function to test the sorting behavior of contact query._

Technical Details
----------------------------------------
_None for the moment._

Comments
----------------------------------------
_Test will need review for best coding practices._

syxys1 committed:
Update ContactGetTest.php

Add a test to verify contact query with results order by contact_sub_type:label.

syxys1 committed:
Merge pull request #1 from syxys1/syxys1-patch-1

Update ContactGetTest.php

colemanw commented:
@syxys1 please remove the `print` statements as our tests don't support them.
There are a few other code style issues, you can view them by clicking on the "details" button next to **CiviCRM-Core-Style** above.

syxys1 committed:
Update ContactGetTest.php

Corrected array syntax, modified variable names to start with lower case and disabled all print statement

syxys1 commented:
@coleman Did the asked corrections and commit changes. Is there anything I got to do to submit again the PR?

colemanw commented on tests/phpunit/api/v4/Action/ContactGetTest.php:
```diff
@@ -21,13 +21,15 @@
 /**
  * @group headless
  */
+
```
@syxys1 you added a blank line here. There's no reason to do that, it's not part of your new test.

syxys1 replied:
@coleman I needed to add use Civi\Api4\ContactType; for my test.  I probably added the blank by mistake.

colemanw commented:
Thanks @syxys1 - you can review the remaining style issues by clicking the "Details" link next to **CiviCRM-Core-Style** just below this comment.

syxys1 committed:
Update ContactGetTest.php

Corrected forgotten use of => in array sort function.

syxys1 commented:
Remove last use of notation => that was used in sort array function.

syxys1 committed:
Update ContactGetTest.php

First iteration at correcting styling error.

colemanw commented on tests/phpunit/api/v4/Action/ContactGetTest.php:
```diff
@@ -60,6 +62,138 @@ public function testGetDeletedContacts(): void {
+  public function testGetWithOrderBy(): void {
+ 
+    // Test dataset #1 : Contact sub type name, label
+    $cntctType = [['Membre', '1.Actif'],
+                  ['Proche', '2.Proche'],
+                  ['Soutien', '3.Soutien'],
+                  ['Employ_', '5.Employé'],
+                  ['DCD', '0.Décédé']
+    ];
```
@syxys1 I'm not sure what you are basing your code style on. There are thousands of examples to follow in our test files and they all use the same consistent style. Can you please check other examples and base your style on that?
Our standard is 2-space indents and new arrays on a new line, and a comma on the last line, so this should be formatted like:
```suggestion
    $cntctType = [
      ['Membre', '1.Actif'],
      ['Proche', '2.Proche'],
      ['Soutien', '3.Soutien'],
      ['Employ_', '5.Employé'],
      ['DCD', '0.Décédé'],
    ];
```

syxys1 committed:
Update ContactGetTest.php

Second iteration at styling correction

syxys1 committed:
Update ContactGetTest.php

New round of styling code styling correction...

syxys1 committed:
Update ContactGetTest.php

Found out how to do Civilint locally.  This version past local test.  Let see how it does on the official site.

syxys1 committed:
Update ContactGetTest.php

syxys1 committed:
Update ContactGetTest.php

eileenmcnaughton commented:
I have resurrected this & the related PR into this PR https://github.com/civicrm/civicrm-core/pull/32856
